src/classes/ACCT_IndividualAccounts_TDTM.cls

```diff
@@ -52,570 +48,705 @@ public class ACCT_IndividualAccounts_TDTM extends TDTM_Runnable {
-                                setAccountIdHHToRename.add(oldContacts[i].AccountId);
-                            }
-                        }
+        if (triggerAction == TDTM_Runnable.Action.AfterInsert) {
 
-                        // old HH Accounts that don't get deleted still need to have name redone
-                        if (oldContacts[i].AccountId != null) {
-                            listAccountIdHHToUpdate.add(oldContacts[i].AccountId);
+            //Updates the Primary Contact Field on New Account
+            if (insertWrapper.accIdConIdToUpdate.size() > 0) {
+            	updateAccountPrimaryContact(insertWrapper.accIdConIdToUpdate, dmlWrapper);
+            }
 
-                            //Used to Adv to inject oppor logic
-                            mapContactIdAccountIdOldMoveOpps.put(c.Id, oldContacts[i].AccountId);
-                        }
-                    }
+            //Updates Household Account name
+            if (insertWrapper.accountIdToRename.size() > 0) {
+                UTIL_ACCT_Naming.updateHHAccountsName(insertWrapper.accountIdToRename);
+            }
 
-                    // check for ownership changes
-                    if (c.OwnerId != oldContacts[i].OwnerId) {
-                    	mapContactIdContactOwnerChange.put(c.Id, c);
-                    }
+            TDTM_TriggerHandler.processDML(dmlWrapper, true);
+            updateRecursionFlag();
+        }
+    }
 
-                    // if the contact's last name has changed and it is using admin account model
-                    // update the name of the related admin account
-                    if (c.LastName != oldContacts[i].LastName && c.AccountId != null)
-                    {
-                        listContactsChangedLastName.add(c);
+    /*******************************************************************************************************
+    * @description Wrapper class for handling after update logic.
+    */
+    private class HandleUpdateWrapperLogic {
+        public Map<Id,Id> contactIdAccountIdNewOppsToMove = new Map<Id,Id>();
+        public Map<Id,Id> contactIdAccountIdOldOppsToMove = new Map<Id,Id>();
+        public Map<Id,Contact> contactIdContactOwnerChange = new Map<Id,Contact>();
+        public Set<Id> accountIdToRename = new Set<Id>();
+        public List<Id> accountIdToUpdate = new List<Id>();
+        public List<Contact> contactsChangedLastName = new List<Contact>();
+
+		/*******************************************************************************************************
+    	* @description Processes after update logic.
+    	* @param c is a Contact record from trigger new.
+    	* @param oldContacts is a list of Contacts from trigger old.
+    	* @param triggerAction which trigger event (BeforeInsert, AfterInsert, etc.).
+    	* @param i is an integer counter.
+    	*/
+        private HandleUpdateWrapperLogic(Contact c, Contact oldContacts, TDTM_Runnable.Action triggerAction) {
```

Comment:
Thanks for updating this! Might be less confusing if `oldContacts` was renamed to `oldContact` though